syncopy/tests/synth_data.py

```diff
@@ -29,26 +30,47 @@ def collect_trials(trial_generator):
     If the underlying trial generating function also accepts
     a `samplerate`, forward this directly.
 
+    If the underlying trial generating function also accepts
+    a `seed`, forward this directly. One can pass a single seed,
+    to be used for all trials, or a list/np.ndarray of seeds with
+    len/size equal to nTrials, with one seed per trial. Leaving
+    at the default value, None, will select a random seed each time,
+    (and it will differ between trials).
+
     The default `nTrials=None` is the identity wrapper and
     just returns the output of the trial generating function
     directly, so a single trial :class:`numpy.ndarray`.
     """
 
     @functools.wraps(trial_generator)
-    def wrapper_synth(nTrials=None, samplerate=1000, **tg_kwargs):
+    def wrapper_synth(nTrials=None, samplerate=1000, seed=None, seed_per_trial=True, **tg_kwargs):
+
+        use_seed_per_trial = False
+        if nTrials is not None and seed is not None and seed_per_trial:  # Use the single seed to create one seed per trial.
+            rng = np.random.default_rng(seed)
+            seed = rng.integers(1000000, size=nTrials)
+            use_seed_per_trial = True
 
         # append samplerate parameter if also needed by the generator
         if 'samplerate' in signature(trial_generator).parameters.keys():
             tg_kwargs['samplerate'] = samplerate
 
-        # do nothing
+        # do nothing (may pass on the scalar seed if the function supports it)
         if nTrials is None:
+            if 'seed' in signature(trial_generator).parameters.keys():
+                tg_kwargs['seed'] = seed
             return trial_generator(**tg_kwargs)
         # collect trials
         else:
             trl_list = []
 
-            for _ in range(nTrials):
+            for trial_idx in range(nTrials):
+                if 'seed' in signature(trial_generator).parameters.keys():
+                    if use_seed_per_trial:
+                        tg_kwargs['seed'] = seed[trial_idx]
+                    else:
+                        tg_kwargs['seed'] = seed
```

Comment:
with this we just generate identical trial copies right? I don't think there is any real use case?!

Reply:
Yes, with `use_seed_per_trial=False` you get the old behaviour (before this PR) that all trials are equal. Plus you can, optionally, make sure they are equal and fixed (by giving a seed that is different from `None`).

Reply:
Mmh, so the old behaviour was not equal trials right? Otherwise, many of our tests would make little sense.. But completely equal trials make only very little sense imho. We can leave it like that, but I expect that we'll never use this code branch